src/server/database/database_updater.py

```diff
@@ -0,0 +1,54 @@
+from datetime import date, timedelta
+from typing import Optional
+
+import requests
+
+from .. import QUERIES
+from . import DatabaseProvider
+
+
+class DatabaseUpdater:
+    """Class for updating the database with new stock matket data."""
+
+    @staticmethod
+    def daily_database_update() -> None:
+        """Update the database with prices up to the current day."""
+        today_date: date = date.today()
+        last_known_date: date = DatabaseUpdater.check_last_known_date()
+        print(f"DEBUG: Daily database update triggered. Today is {today_date}.")
+
+        if today_date == last_known_date:
+            print("DEBUG: Nothing to update.")
+
+        while last_known_date < today_date:
+            current_date: date = last_known_date + timedelta(days=1)
+            print(f"DEBUG: Updating price for {current_date}.")
+            DatabaseUpdater.update_selected_date(current_date)
+            last_known_date = DatabaseUpdater.check_last_known_date()
+
+    @staticmethod
+    def check_last_known_date() -> date:
```

Comment:
I think that this method is not used anywhere outside of this class, please mark it as private (prepend `_` to the name).

Reply:
done